### PR TITLE
Fix debugserver's qProcessInfo reporting of maccatalyst binaries

### PR DESCRIPTION
**Pull request by adrian-prantl**
This patch is similar in spirit to https://reviews.llvm.org/D84480,
but does the maccatalyst/macosx disambiguation. I also took the
opportunity to factor out the gdb-remote packet log scanning used by
several testcases into lldbutil functions.

rdar://problem/66059257

Differential Revision: https://reviews.llvm.org/D84576

(cherry picked from commit 7e9bab6ad51af1cab0c7457e4323166af3ac797c)

 Conflicts:
	lldb/test/API/macosx/macabi/TestMacABImacOSFramework.py

### CODE REVIEW

**Comment by adrian-prantl** [comment]
@swift-ci test

File: lldb/tools/debugserver/source/MacOSX/MachProcess.mm [hunk 1]
```diff
@@ -794,37 +820,21 @@ static bool FBSAddEventDataToOptions(NSMutableDictionary *options,
...
-        inf.min_version_os_version = GetMacCatalystVersionString();
-      } else if (dyld_platform != PLATFORM_MACCATALYST &&
-                 inf.min_version_os_name == "macosx") {
-        // This is a binary with both PLATFORM_MACOS and
-        // PLATFORM_MACCATALYST load commands and the process is not
-        // running as PLATFORM_MACCATALYST. Stick with the
-        // "macosx" load command that we've already processed,
-        // ignore this one, which is presumed to be a
+      if (dyld_platform != PLATFORM_MACCATALYST &&
+          inf.min_version_os_name == "macosx") {
```
Comment (fredriss) [inline lldb/tools/debugserver/source/MacOSX/MachProcess.mm:827]: What happens if the catalyst load command is the first one?

Comment (adrian-prantl) [reply lldb/tools/debugserver/source/MacOSX/MachProcess.mm:827]: IIRC, the the `if` condition implies that we must have already processed a non-catalyst load command.

Comment (adrian-prantl) [reply lldb/tools/debugserver/source/MacOSX/MachProcess.mm:827]: `inf.min_version_os_name` is from a previous iteration of the loop.

Comment (fredriss) [reply lldb/tools/debugserver/source/MacOSX/MachProcess.mm:827]: I misread the patch. I though it inverted the condition, but it just folded the first `if` into  `GetDeploymentInfo`. This condition was already there. This LGTM

**Pull request merged**